### PR TITLE
Integrate Fetch Project  and Add Provider to Add Project Page.

### PR DESCRIPTION
**Pull request by chevyB**
Issue: https://app.asana.com/0/1202107027507930/1202139395534041/f

### Services to RUN
- FE
- BFF
- Backlog

### NOTE

- ignore warning for now in front-end unit testing
- Also note that I will refactor the unit test on my next task. So if you have any concern regarding on backlog service unit testing, pls add it here:

https://app.asana.com/0/1202107027507930/1202266969657290/f (edited)

### CODE REVIEW

**Review by NielmarLaurente — APPROVED** [review]
## Approved PR #25 
### Working 👍 
- Add new Project Page is created
- Can add new Provider with API Key
- Validate API Key and Selected provider
- Frontend and Backlog Test Passed

<details>

<summary> Screenshot </summary>

image

image

image

image

</details>

**Testing Approval time : 0.5 hr
Total comments : 0
Total Iterations : 0**

File: backlog/test/controllers/ProviderController.test.ts [hunk 1]
```diff
@@ -1,60 +1,82 @@
...
+  let res: TypedResponse
+
+  const mockAxios = new MockAdapter(axios)
+
+  const payload = {
+    user_id: 1,
+    name: 'Backlog',
+    api_key: '123apikey123',
+    project_id: faker.datatype.number()
+  }
```
Comment (escbooster12-dev) [inline backlog/test/controllers/ProviderController.test.ts:27]: - this data persist in to actual database :thinking: was this already from previous PR?
image

Comment (chevyB) [reply backlog/test/controllers/ProviderController.test.ts:27]: Yes, and as mentioned in slack.  that I will refactor the unit test on my next task. So if you have any concern regarding on backlog service unit testing, pls add it here:
https://app.asana.com/0/1202107027507930/1202266969657290/f

Comment (escbooster12-dev) [reply backlog/test/controllers/ProviderController.test.ts:27]: i updated the asana, please check if there are lacking :man:

**Review by escbooster12-dev — COMMENTED** [review]
I encounter an error. while visiting add project page. 
image

**Review by escbooster12-dev — APPROVED** [review]
**Working functionalities**
- verify api key to backlog
- fetch projects from backlog
- select and insert project from backlog

**Notes**
- there are warnings in frontend that should need to be fixed
- select option in _add project page_ needs to double click in order to work (should be fix in the future)
- some of the test in backlog persisted in the actual database. (should be mock in the future)

**Updated the separted task asana:** https://app.asana.com/0/1202107027507930/1202266969657290/f

image

**Pull request merged**